api/src/test/java/org/openmrs/module/fhir2/api/FhirImmunizationServiceTest.java

```diff
@@ -0,0 +1,262 @@
+		Immunization newImmunization = parser.parseResource(Immunization.class, "{\n"
+		        + "  \"resourceType\": \"Immunization\",\n" + "  \"status\": \"completed\",\n" + "  \"vaccineCode\": {\n"
+		        + "    \"coding\": [\n" + "      {\n" + "        \"code\": \"15f83cd6-64e9-4e06-a5f9-364d3b14a43d\",\n"
+		        + "        \"display\": \"Aspirin as a vaccine\"\n" + "      }\n" + "    ]\n" + "  },\n"
+		        + "  \"patient\": {\n" + "    \"reference\": \"Patient/a7e04421-525f-442f-8138-05b619d16def\",\n"
+		        + "    \"type\": \"Patient\"\n" + "  },\n" + "  \"encounter\": {\n"
+		        + "    \"reference\": \"Encounter/7d8c1980-6b78-11e0-93c3-18a905e044dc\",\n"
+		        + "    \"type\": \"Encounter\"\n" + "  },\n" + "  \"occurrenceDateTime\": \"2020-07-08T18:30:00.000Z\",\n"
+		        + "  \"manufacturer\": {\n" + "    \"display\": \"Acme\"\n" + "  },\n" + "  \"lotNumber\": \"FOO1234\",\n"
+		        + "  \"expirationDate\": \"2022-07-31T18:30:00.000Z\",\n" + "  \"performer\": [\n" + "    {\n"
+		        + "      \"actor\": {\n" + "        \"reference\": \"Practitioner/f9badd80-ab76-11e2-9e96-0800200c9a66\",\n"
+		        + "        \"type\": \"Practitioner\"\n" + "      }\n" + "    }\n" + "  ],\n" + "  \"protocolApplied\": [\n"
+		        + "    {\n" + "      \"doseNumberPositiveInt\": 2,\n" + "      \"series\": \"Dose 2\"\n" + "    }\n"
+		        + "  ]\n" + "}");
+		
+		// replay
+		Immunization savedImmunization = service.create(newImmunization);
+		Obs obs = obsService.getObsByUuid(savedImmunization.getId());
+		
+		// verify
+		ImmunizationObsGroupHelper helper = new ImmunizationObsGroupHelper(conceptService);
+		helper.validateImmunizationObsGroup(obs);
+		assertObsCommons(obs, "a7e04421-525f-442f-8138-05b619d16def", "7d8c1980-6b78-11e0-93c3-18a905e044dc",
+		    "f9badd80-ab76-11e2-9e96-0800200c9a66");
+		
+		obs.getGroupMembers().forEach(o -> {
+			assertObsCommons(o, "a7e04421-525f-442f-8138-05b619d16def", "7d8c1980-6b78-11e0-93c3-18a905e044dc",
+			    "f9badd80-ab76-11e2-9e96-0800200c9a66");
+		});
+		
+		Map<String, Obs> members = helper.getObsMembersMap(obs);
+		assertThat(members.get(ciel984).getValueCoded().getUuid(), is("15f83cd6-64e9-4e06-a5f9-364d3b14a43d"));
+		assertThat(members.get(ciel1410).getValueDatetime(),
+		    equalTo(new DateTimeType("2020-07-08T18:30:00.000Z").getValue()));
+		assertThat(members.get(ciel1418).getValueNumeric(), equalTo(2.0));
+		assertThat(members.get(ciel1419).getValueText(), is("Acme"));
+		assertThat(members.get(ciel1420).getValueText(), is("FOO1234"));
+		assertThat(members.get(ciel165907).getValueDatetime(),
+		    equalTo(new DateTimeType("2022-07-31T18:30:00.000Z").getValue()));
+	}
+	
+	@Test
+	public void updateImmunization_shouldUpdateImmunizationAccordingly() throws ParseException {
+		
+		// setup
+		FhirContext ctx = FhirContext.forR4();
+		IParser parser = ctx.newJsonParser();
+		Immunization updatedImmunization = parser.parseResource(Immunization.class,
```

Comment:
I'd tend to prefer this being externalised in a file, but there's no real harm.

Reply:
I hear you, however for the ease of debugging I'll currently leave those to not be in test resources, if that's ok?